### PR TITLE
Added Retina Image widget

### PR DESCRIPTION
**Pull request by sushmak02**

### CODE REVIEW

**Comment by Nikschavan** [comment]
@sushmak02 / @ahemads - Can you also check the PHPCS recommendations in Travis?

File: assets/css/header-footer-elementor.css [hunk 1]
```diff
@@ -28,6 +28,6 @@
 
 /* Fix: Header hidden below the page content */
 .ehf-header #masthead {
-	z-index: 200;
+	z-index: 30;
```
[the remark this answers is not in the document]
Comment (sushmak02) [reply assets/css/header-footer-elementor.css]: We didn't change this. Maybe this is done by mistake. I will revert the unrelated changes.

File: inc/class-header-footer-elementor.php [hunk 2]
```diff
@@ -327,18 +376,18 @@ public function render_template( $atts ) {
...
-		$id = ! empty( $atts['id'] ) ? apply_filters( 'hfe_render_template_id', intval( $atts['id'] ) ) : '';
+		$id = ! empty( $atts['id'] ) ? intval( $atts['id'] ) : '';
 
 		if ( empty( $id ) ) {
 			return '';
 		}
 
-		if ( class_exists( '\Elementor\Core\Files\CSS\Post' ) ) {
-			$css_file = new \Elementor\Core\Files\CSS\Post( $id );
-		} elseif ( class_exists( '\Elementor\Post_CSS_File' ) ) {
```
[the remark this answers is not in the document]
Comment (sushmak02) [reply inc/class-header-footer-elementor.php]: We didn't change this too. Maybe this is done by mistake. I will revert the unrelated changes.

File: widgets/class-retina.php [hunk 3]
```diff
@@ -0,0 +1,742 @@
...
+}
+
+/**
+ * Elementor Hello World
+ *
+ * Elementor widget for hello world.
+ *
+ * @since x.x.x
+ */
+class Retina extends Widget_Base {
```
[the remark this answers is not in the document]
Comment (Nikschavan) [reply widgets/class-retina.php]: In this PR you can namespace entire new `widgets` directory along with the loader.

I will create another PR where rest of the code can be changed similar to this.

File: widgets/class-retina.php [hunk 4]
```diff
@@ -0,0 +1,742 @@
...
+	 * @since x.x.x
+	 * @param array $settings returns the caption.
+	 *
+	 * @return string
+	 */
+	private function get_caption( $settings ) {
+		$caption = '';
+		if ( ! empty( $settings['caption_source'] ) ) {
+
+			if ( 'custom' == $settings['caption_source'] ) {
```
[the remark this answers is not in the document]
Comment (sushmak02) [reply widgets/class-retina.php]: Done!

File: assets/css/header-footer-elementor.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (sushmak02) [reply assets/css/header-footer-elementor.css]: Done!

File: inc/class-header-footer-elementor.php [hunk 5]
```diff
@@ -66,6 +66,12 @@ function __construct() {
 
 			add_shortcode( 'hfe_template', array( $this, 'render_template' ) );
 
+			// Register widgets.
+			add_action( 'elementor/widgets/widgets_registered', array( $this, 'register_widgets' ) );
```
[the remark this answers is not in the document]
Comment (sushmak02) [reply inc/class-header-footer-elementor.php]: Done!

File: inc/class-header-footer-elementor.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (sushmak02) [reply inc/class-header-footer-elementor.php]: Done!

File: widgets/class-retina.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (sushmak02) [reply widgets/class-retina.php]: Done!

File: inc/class-header-footer-elementor.php [hunk 6]
```diff
@@ -66,6 +66,8 @@ function __construct() {
 
 			add_shortcode( 'hfe_template', array( $this, 'render_template' ) );
 
+			require HFE_DIR . 'inc/widgets-manager/class-hfe-widgets-loader.php';
```
Comment (Nikschavan) [inline inc/class-header-footer-elementor.php]: include the file from the method `includes()`

Comment (sushmak02) [reply inc/class-header-footer-elementor.php]: @Nikschavan Done. Please check

**Pull request merged**